src/controls/controls.actions.ts

```diff
@@ -268,138 +270,116 @@ import { renderCircleControl, renderSquareControl } from './controls.render';
   }
 
   /**
-   * Detect if the fabric object is flipped on one side.
-   * @param {fabric.Object} target
-   * @return {Boolean} true if one flip, but not two.
+   * @see https://github.com/fabricjs/fabric.js/pull/8380
    */
-  function targetHasOneFlip(target) {
-    return target.flipX !== target.flipY;
-  }
-
-  /**
-   * Utility function to compensate the scale factor when skew is applied on both axes
-   * @private
-   */
-  function compensateScaleForSkew(
-    target,
-    oppositeSkew,
-    scaleToCompensate,
-    axis,
-    reference
+  function skewObject(
+    axis: TAxis,
+    { target, ex, ey, originX, originY, skewX, skewY, flipX, flipY },
+    pointer: Point
   ) {
-    if (target[oppositeSkew] !== 0) {
-      var newDim = target._getTransformedDimensions()[axis];
-      var newValue = (reference / newDim) * target[scaleToCompensate];
-      target.set(scaleToCompensate, newValue);
-    }
+    const { counterAxis, skew: skewKey } = AXIS_KEYS[axis],
+      offset = pointer
+        .subtract(new Point(ex, ey))
+        .divide(new Point(target.scaleX, target.scaleY))[axis],
+      flip = new Point(flipX ? -1 : 1, flipY ? -1 : 1)[axis],
+      skewingStart = new Point(skewX, skewY)[axis],
+      skewingBefore = new Point(target.skewX, target.skewY)[axis],
+      shearingStart = Math.tan(degreesToRadians(skewingStart)),
+      counterOriginFactor = new Point(
+        target.resolveOriginX(originX),
+        target.resolveOriginY(originY)
+      )[counterAxis],
+      offsetFactor = -Math.sign(counterOriginFactor) * flip * 2,
+      // let a, b be the size of target
+      // let a' be the value of a after applying skewing
+      // then:
+      // a' = a + b * skewA => skewA = (a' - a) / b
+      // the value b is tricky since skewY is applied before skewX
+      b =
+        axis === 'y'
+          ? target._getTransformedDimensions({
+              scaleX: 1,
+              scaleY: 1,
+              // since skewY is applied before skewX b (=width) is not affected by skewX in this case
```

Comment:
is this straight forward enough?
can you understand it?

Reply:
so if this is the fix, what are all the other diffs about?